.gitignore

```diff
@@ -1,5 +1,12 @@
-build
-dist
+bazel-*
+# Bazel's MODULE lockfile isn't ready to check in yet as of Bazel 7.1.
```

Comment:
nit: some of these likely fixed in Bazel 8 - but I'm still fine not checking in the lockfile, I don't think it buys us anything